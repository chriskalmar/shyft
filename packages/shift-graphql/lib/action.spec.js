
import {
  Action,
  DataTypeString,
  DataTypeInteger,
} from 'shift-engine';

import {
  generateActionDataInput,
  generateActionInput,
  generateActionDataOutput,
  generateActionOutput,
} from './action';


describe('action', () => {

  const simpleAction = new Action({
    name: 'newSimpleAction',
    description: 'do something',
    input: {
      firstName: {
        type: DataTypeString,
        description: 'First name',
      },
      lastName: {
        type: DataTypeString,
        description: 'Last name',
        required: true,
      },
    },
    output: {
      luckyNumber: {
        type: DataTypeInteger,
        description: 'The perfect lucky number for the given name'
      }
    },
    resolve() { },
  })


  it('should generate an action data input type', () => {

    const type = generateActionDataInput(simpleAction)
    expect(type.name).toEqual('NewSimpleActionDataInput');

    const fields = type.getFields()
    expect(fields).toMatchSnapshot();
  })


  it('should generate an action input type', () => {

    const inputDataInputType = generateActionDataInput(simpleAction)
    const type = generateActionInput(simpleAction, inputDataInputType)
    expect(type.name).toEqual('NewSimpleActionInput');

    const fields = type.getFields()
    expect(fields).toMatchSnapshot();
  })


  it('should generate an action data output type', () => {

    const type = generateActionDataOutput(simpleAction)
    expect(type.name).toEqual('NewSimpleActionDataOutput');

    const fields = type.getFields()
    expect(fields).toMatchSnapshot();
  })


  it('should generate an action output type', () => {

    const outputDataOutputType = generateActionDataOutput(simpleAction)
    const type = generateActionOutput(simpleAction, outputDataOutputType)
    expect(type.name).toEqual('NewSimpleActionOutput');

    const fields = type.getFields()
    expect(fields).toMatchSnapshot();
  })


})
