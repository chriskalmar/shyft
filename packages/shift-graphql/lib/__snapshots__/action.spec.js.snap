// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`action should generate an action data input type 1`] = `
Object {
  "firstName": Object {
    "description": "First name",
    "name": "firstName",
    "type": "String",
  },
  "lastName": Object {
    "description": "Last name",
    "name": "lastName",
    "type": "String!",
  },
}
`;

exports[`action should generate an action data output type 1`] = `
Object {
  "luckyNumber": Object {
    "args": Array [],
    "description": "The perfect lucky number for the given name",
    "isDeprecated": false,
    "name": "luckyNumber",
    "type": "Int",
  },
}
`;

exports[`action should generate an action input type 1`] = `
Object {
  "clientMutationId": Object {
    "name": "clientMutationId",
    "type": "String",
  },
  "data": Object {
    "name": "data",
    "type": "NewSimpleActionDataInput!",
  },
}
`;

exports[`action should generate an action output type 1`] = `
Object {
  "clientMutationId": Object {
    "args": Array [],
    "isDeprecated": false,
    "name": "clientMutationId",
    "type": "String",
  },
  "result": Object {
    "args": Array [],
    "isDeprecated": false,
    "name": "result",
    "type": "NewSimpleActionDataOutput",
  },
}
`;
